Client/mods/deathmatch/logic/luadefs/CLuaVehicleDefs.cpp

```diff
@@ -4340,3 +4342,19 @@ bool CLuaVehicleDefs::SpawnVehicleFlyingComponent(CClientVehicle* const vehicle,
 
     return vehicle->SpawnFlyingComponent(partNodeIndex, collisionType, removalTime.value_or(-1));
 }
+
+bool CLuaVehicleDefs::SetSmokeTrailEnabled(CClientVehicle* vehicle, bool state) noexcept
+{
+    std::uint16_t model = vehicle->GetModel();
+    if (model != 512 && model != 513)
+        return false;
```

Comment:
Does this function work with `engineRequestModel( "vehicle", 512 )` ?

Reply:
the model id is invalid, so what do you suggest by checking what

Reply:
Does this function work with allocated vehicles if you remove the model check?

Reply:
no it does not

Reply:
sad